lib/src/main/java/io/ably/lib/http/AsyncHttp.java

```diff
@@ -147,6 +148,8 @@ private AblyRequestWithFallback(
 		public void run() {
 			int retryCountRemaining = Hosts.isRestFallbackSupported(http.host) ? http.options.httpMaxRetryCount : 0;
 			String candidateHost = http.host;
+			if(http.options.fallbackHosts != null)
+				Collections.shuffle(http.options.fallbackHosts);
```

Comment:
So was Piotr wrong? Is there any situation where someone overwrites the public field and we need to ensure the new value gets randomised?